src/DocBlock/TypeExpression.php

```diff
@@ -454,6 +487,49 @@ private function parseCommaSeparatedInnerTypes(int $startIndex, string $value):
         }
     }
 
+    private function parseCallableTemplateInnerTypes(int $startIndex, string $value): void
```

Comment:
I would really like to see some comments above/inside this method, because it's not clear for me what happens here, even though I feel pretty familiar with regexes (e.g. not sure what `\G` does here, as I only know what it does in general). I don't get the `$addedPrefix.$consumedValue.'>(): void'` part too. Seems too magical and may be hard to maintain without some information about the the `$value` and logic that process it.

Reply:
the magical regex is to parse inner callable type

Reply:
`\G` is documented in https://www.php.net/manual/en/regexp.reference.escape.php